tests/client/amo/containers/TestApp.js

```diff
@@ -17,15 +17,44 @@ describe('App', () => {
         return <p>The component</p>;
       }
     }
-    const root = findRenderedComponentWithType(renderIntoDocument(
-      <I18nProvider i18n={getFakeI18nInst()}>
-        <App>
-          <MyComponent />
-        </App>
-      </I18nProvider>), App).getWrappedInstance();
+    const i18n = getFakeI18nInst();
+    const root = renderIntoDocument(
+      <AppBase i18n={i18n} isAuthenticated>
+        <MyComponent />
+      </AppBase>
+      );
 
     const rootNode = findDOMNode(root);
     assert.equal(rootNode.tagName.toLowerCase(), 'div');
     assert.equal(rootNode.querySelector('p').textContent, 'The component');
   });
+
+  it('shows a log in button', () => {
+    const i18n = getFakeI18nInst();
+    const handleLogIn = sinon.spy();
+    const location = sinon.stub();
+    const root = renderIntoDocument(
+      <AppBase i18n={i18n} isAuthenticated={false} handleLogIn={handleLogIn} location={location} />
+    );
+    const button = root.logInButton;
+    assert.equal(button.textContent, 'Log in');
+    Simulate.click(button);
+    assert.ok(handleLogIn.calledWith(location));
+  });
+
+  it('tells you if you are logged in', () => {
+    const i18n = getFakeI18nInst();
+    const root = renderIntoDocument(<AppBase i18n={i18n} isAuthenticated />);
+    assert.equal(findDOMNode(root).querySelector('footer').textContent, 'You are logged in');
+  });
+
+  it('updates the location on handleLogIn', () => {
+    const _window = { location: '/foo' };
+    const location = { pathname: '/bar', query: { q: 'wat' } };
+    const startLoginUrlStub = sinon.stub(api, 'startLoginUrl').returns('https://a.m.org/login');
```

Comment:
The sandbox should save us, yeah. The nice thing about sinon is it will fail if the method does not exist. This mostly is annoying but could save us if we rename the method (not if we change it's functionality though, unfortunately).